### PR TITLE
Test

### PR DESCRIPTION
**Pull request by ThibaultLacharme**

### CODE REVIEW

File: src/test_calculator.hpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#ifndef TEST_CALCULATOR_HPP
```
Comment (martinRenou) [inline src/test_calculator.hpp]: I guess you did not mean to commit this file, it is in `src/`

**Review by martinRenou — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -21,6 +19,10 @@ configure_file (
 set(xeus_REQUIRED_VERSION 0.19.1)
 set(cppzmq_REQUIRED_VERSION 4.3.0)
 
+#set(XEUS_CALC_HEADERS
+#    ${CMAKE_CURRENT_SOURCE_DIR}/src/xeus_calc_interpreter.hpp
+#    )
```
Comment (JohanMabille) [inline CMakeLists.txt]: Please do not commit commented code

File: CMakeLists.txt [hunk 3]
```diff
@@ -42,15 +44,35 @@ endif()
...
+    set(BUILD_TESTS ON)
+endif()
+
+
+# take into account the CMakeList in test
+if(BUILD_TESTS)
+    add_subdirectory(test)
+endif()
+
 # My kernel executable
```
Comment (JohanMabille) [inline CMakeLists.txt]: This comment is now irrelevant

File: src/xeus_calc_interpreter.cpp [hunk 4]
```diff
@@ -63,42 +63,42 @@ namespace xeus_calc
                 continue;
             }
 
-            char c = token[0];
-            size_t idx = ops.find(c);
+            char c = token[0]; // study the first character of the token at each loop
```
Comment (JohanMabille) [inline src/xeus_calc_interpreter.cpp]: study => checks

File: test/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+endif ()
+
+if(NOT CMAKE_BUILD_TYPE)
+    message(STATUS "Setting tests build type to Release")
+    set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE)
+else()
+    message(STATUS "Tests build type is ${CMAKE_BUILD_TYPE}")
+endif()
+
+#include(CheckCXXCompilerFlag)
```
Comment (JohanMabille) [inline test/CMakeLists.txt]: to be removed

File: test/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+string(TOUPPER "${CMAKE_BUILD_TYPE}" U_CMAKE_BUILD_TYPE)
+
+#include(set_compiler_flag.cmake)
```
Comment (JohanMabille) [inline test/CMakeLists.txt]: same remark

File: src/xeus_calc_interpreter.cpp [hunk 7]
```diff
@@ -108,71 +108,71 @@ namespace xeus_calc
...
             {
-                publish_stream("stdout","Push\t\t");
-                stack.push_back(tokenNum);
+                publish("stdout","Push\t\t");
+                evaluation.push_back(tokenNum);
             }
             else
             {
-                if (stack.size() >= 2)
+                if (evaluation.size() >= 2) // error management, checks for missing operand
```
Comment (JohanMabille) [inline src/xeus_calc_interpreter.cpp]: Comment not really meaningful

File: src/xeus_calc_interpreter.cpp [hunk 8]
```diff
@@ -108,71 +108,71 @@ namespace xeus_calc
             }
             else
             {
-                ss << token << ' ';
+                parsed_infix << token << ' '; // if the first value of the token is the number, we can assume that the whole token is composed of numbers and thus push it
```
Comment (JohanMabille) [inline src/xeus_calc_interpreter.cpp]: Comments should be placed before the code

**Review by JohanMabille — CHANGES_REQUESTED** [review]

**Review by JohanMabille — APPROVED** [review]

**Pull request merged**